pallets/author-noting/src/benchmarks.rs

```diff
@@ -57,31 +57,45 @@ benchmarks! {
         // Depend on the number of parachains registered
         let x in 0..100;
 
-        let mut sproof_builder = test_sproof::ParaHeaderSproofBuilder::default();
         let mut container_chains = vec![];
 
-        for para_id in 0..x {
-            let para_id = para_id.into();
-            container_chains.push(para_id);
-            // Mock assigned authors for this para id
-            let author: T::AccountId = account("account id", 0u32, 0u32);
-            // Use the max allowed value for num_each_container_chain
-            let num_each_container_chain = 2;
-            T::ContainerChainAuthor::set_authors_for_para_id(para_id, vec![author; num_each_container_chain]);
-            sproof_builder.num_items += 1;
-        }
-
-        let (root, proof) = sproof_builder.into_state_root_and_proof();
-
-        let data = tp_author_noting_inherent::OwnParachainInherentData {
-            relay_storage_proof: proof,
+        let data = if TypeId::of::<<<T as Config>::RelayOrPara as RelayOrPara>::InherentArg>() == TypeId::of::<tp_author_noting_inherent::OwnParachainInherentData>() {
+            let mut sproof_builder = test_sproof::ParaHeaderSproofBuilder::default();
+
+            for para_id in 0..x {
+                let para_id = para_id.into();
+                container_chains.push(para_id);
+                // Mock assigned authors for this para id
+                let author: T::AccountId = account("account id", 0u32, 0u32);
+                // Use the max allowed value for num_each_container_chain
+                let num_each_container_chain = 2;
+                T::ContainerChainAuthor::set_authors_for_para_id(para_id, vec![author; num_each_container_chain]);
+                sproof_builder.num_items += 1;
+            }
+
+            let (root, proof) = sproof_builder.into_state_root_and_proof();
+
+            T::RelayOrPara::set_current_relay_chain_state(cumulus_pallet_parachain_system::RelayChainState {
+                state_root: root,
+                number: 0,
+            });
+
+            let arg = tp_author_noting_inherent::OwnParachainInherentData {
+                relay_storage_proof: proof,
+            };
+
+            *(Box::new(arg) as Box<dyn Any>).downcast().unwrap()
+        } else if TypeId::of::<<<T as Config>::RelayOrPara as RelayOrPara>::InherentArg>() == TypeId::of::<()>() {
+            // TODO: we need to write to storage the elements from the proof
+            unimplemented!("Benchmark will fail because headers have not been written to storage");
```

Comment:
Ok this should not be too difficult to do if we are able to configure this for a runtime that has `parachains_common` installed. else we directly put in storage with the HEAD_KEYS

Reply:
The main problem here is that the benchmark script cannot create proofs and so it needs to use the `mock_proof`, but the `mock_proof` only works when you need a storage proof. I could have added a second mock_proof or duplicated the logic to support relay mode here, but I won't be able to test it until we support benchmarks in starlight.

Reply:
Alright